ISHPullUp/ISHPullUpViewController.m

```diff
@@ -114,19 +114,25 @@ - (void)viewWillLayoutSubviews {
     }
 }
 
-#pragma mark Gestures
+- (void)setLocked:(BOOL)locked {
+    _locked = locked;
+    self.panGesture.enabled = !locked;
+}
+
+#pragma mark Pan Gesture
 
-- (void)setupGestureRecognizerForViewController:(UIViewController *)vc {
+- (void)setupPanGestureRecognizerForViewController:(UIViewController *)vc {
     // remove previous gesture recognizer if needed
     [self.panGesture.view removeGestureRecognizer:self.panGesture];
 
     // setup new gesture recognizer on vc's view
-    self.panGesture = [[UIPanGestureRecognizer alloc] initWithTarget:self action:@selector(handleGesture:)];
-    self.panGesture.delegate = self;
-    [vc.view addGestureRecognizer:self.panGesture];
+    UIPanGestureRecognizer *panGesture = [[UIPanGestureRecognizer alloc] initWithTarget:self action:@selector(handlePanGesture:)];
+    self.panGesture = panGesture;
```

Comment:
Should we add `self.panGesture.enabled = !self.locked;` in case the view controller is locked before view did load?